src/pages/patientView/clinicalInformation/PatientViewPageStore.spec.ts

```diff
@@ -33,4 +35,36 @@ describe('ClinicalInformationSamplesTable', () => {
         assert.deepEqual(result,[]);
     });
 
+    it('won\'t fetch cosmic data if there are no mutations', ()=>{
+
+        const fetchStub = sinon.stub();
+
+        let mockInstance = {
+            mutationData: { result:[] },
+            internalClient: {
+                fetchCosmicCountsUsingPOST: fetchStub
+            }
+        };
+
+        store.cosmicDataInvoke.apply(mockInstance).then((data: any)=>{
+           assert.isUndefined(data);
+           assert.isFalse(fetchStub.called);
+        });
+
+    });
+
+    it('won\'t fetch onkokb data if there are no mutations', ()=>{
+
+        const fetchStub = sinon.stub();
+
+        let mockInstance = {
+            mutationData: { result:[] }
+        };
+
+        store.oncoKbDataInvoke.apply(mockInstance).then((data: any)=>{
```

Comment:
i should be checking to make sure the api client is not called, but this is still pretty good.  to make this easier we need to put api clients on instance of store OR set up a solution called rewire so that we can stub imports of modules.